modules/features2d/src/gftt.cpp

```diff
@@ -106,14 +107,14 @@ class GFTTDetector_Impl CV_FINAL : public GFTTDetector
                 cvtColor( image, grayImage, COLOR_BGR2GRAY );
 
             goodFeaturesToTrack( grayImage, corners, nfeatures, qualityLevel, minDistance, _mask,
-                                blockSize, gradSize, useHarrisDetector, k );
+                                 cornersQuality, blockSize, gradSize, useHarrisDetector, k );
         }
 
-        keypoints.resize(corners.size());
-        std::vector<Point2f>::const_iterator corner_it = corners.begin();
-        std::vector<KeyPoint>::iterator keypoint_it = keypoints.begin();
-        for( ; corner_it != corners.end() && keypoint_it != keypoints.end(); ++corner_it, ++keypoint_it )
-            *keypoint_it = KeyPoint( *corner_it, (float)blockSize );
+        CV_Assert(corners.size() == cornersQuality.size());
+
+        keypoints.clear();
```

Comment:
`.reserve()` is necessary to avoid multiple re-allocations.

Reply:
Fixed.